### PR TITLE
Dish create page

### PR DESCRIPTION
**Pull request by luciademoja**
PT Story: https://www.pivotaltracker.com/story/show/128658187

Changes proposed in this pull request:
- Adds dish class 
- Sets name and price as required fields
- Includes optional fields
- Shows error when trying to save without filling in name and price

What I have learned working on this feature:
Manage the controller and connect the dots between all the files generated when working with a new model.

Screenshots:
skarmavbild 2016-08-26 kl 11 49 24

Ready for review!

### CODE REVIEW

File: spec/models/dish_spec.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Dish, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (tochman) [inline spec/models/dish_spec.rb]: A workflow comment: At this point you should have written specs for the Dish model.

**Comment by luciademoja** [comment]
This needs to wait, I forgot to work on the last task before doing submitting the PR
Sorry 👎

File: config/routes.rb [hunk 2]
```diff
@@ -5,4 +5,5 @@
   resources :restaurants, only: [:new, :create]
   root controller: :landing, action: :index
   resources :menus, only: [:new, :create]
+  resources :dishes, only: [:new, :create]
```
Comment (tochman) [inline config/routes.rb]: This route should have been created when needed, meaning that there should be an acceptance test that tries to access it before you actually create it.

File: app/models/dishes.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Dishes < ApplicationRecord
+  belongs_to :menu
+  validates_presence_of :name,
+                        :price,
+
+end
```
Comment (tochman) [inline app/models/dishes.rb]: At this point you should definitely have specs for Dish

File: spec/factories/dishes.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+FactoryGirl.define do
+  factory :dish do
+    ActiveRecord ""
```
Comment (tochman) [inline spec/factories/dishes.rb]: ??

File: spec/models/dish_spec.rb (hunk 1 above)
Comment (tochman) [reply spec/models/dish_spec.rb]: Also, why do we need a Dish model at this stage at all?

File: app/models/dishes.rb [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Dishes < ApplicationRecord
```
Comment (tochman) [inline app/models/dishes.rb]: Dishes?? Should be Dish

File: app/models/dishes.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+  belongs_to :menu
+  validates_presence_of :name,
+                        :price,
```
Comment (tochman) [inline app/models/dishes.rb]: The last comma?? Should throw you an error and a spec would have exposed ut ;-)

File: features/create_dishes.feature [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  Scenario: Create a list of dishes offered
+    Given I am on the "create dishes" page
+    And show me the page
+    And I fill "Name" with "Dish Name"
+    And I fill "Price" with "3.00"
+    And I fill "Allergy Information" with "Peanuts"
+    And I fill "Incredients" with "Mince"
+    And I fill "Calories" with "429"
+    And I click "Create"
+    Then I should be on the "home" page
```
Comment (tochman) [inline features/create_dishes.feature:19]: Good thing. Next time START with that

File: app/views/dishes/new.html.haml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+    = f.text_field :name
+  .field
+    = f.label :price
+    = f.text_field :price
+  .field
+    = f.label :allergy_info
+    = f.text_area :allergy_info
+    .field
+      = f.label :price
+      = f.text_field :price
```
Comment (tochman) [inline app/views/dishes/new.html.haml]: Redundant? Two fields for :price?

File: app/controllers/dishes_controller.rb [hunk 9]
```diff
@@ -0,0 +1,23 @@
+class DishesController < ApplicationController
+  def new
+    @dish = Dish.new
+  end
+
+  # def create
+  #   @menu = Dishes.new(dish_params)
```
Comment (tochman) [inline app/controllers/dishes_controller.rb]: Cut and paste is not cool ;-)

File: app/controllers/dishes_controller.rb [hunk 10]
```diff
@@ -3,21 +3,21 @@ def new
...
-  #
-  # def dish_params
-  #   params.require(:dish).permit(:name, :price)
-  # end
+  def create
+    @dish = Dish.new(dish_params)
+    if @dish.save
+      redirect_to root_path
+    else
+      flash[:error] = 'A name is required'
```
Comment (tochman) [inline app/controllers/dishes_controller.rb]: I would use the validation messages from the model. You can access them by: 

```
@dish.errors.full_messages
```

**Comment by tochman** [comment]
@lollypop27 && @luciademoja Good job on this feature. My comments are mostly about the workflow and not about the content. I just want you to follow the Acceptance-Unit Test Cycle more strict. ;-)

**Comment by diraulo** [comment]
@luciademoja we have a merge conflict here

**Comment by tochman** [comment]
@luciademoja Me and @MikaelFeher worked a bit on this code and created a PR toward your fork of this repo. Have a look at that and merge it in if you are confident with those proposed changes.

File: features/step_definitions/create_menu_page.rb [hunk 11]
```diff
@@ -16,8 +16,17 @@
       path = new_user_registration_path
     when 'restaurants' then
       path = new_restaurant_path
+    when 'create dish' then
```
Comment (diraulo) [inline features/step_definitions/create_menu_page.rb:19]: Is this really necessary? You have a step that visits the create dish page and the path is set there.

Comment (tochman) [reply features/step_definitions/create_menu_page.rb:19]: That step def is a bit messy. Leave this for the refactoring of step defs chore?

Comment (MikaelFeher) [reply features/step_definitions/create_menu_page.rb:19]: Yes, leave it, there is much work to be done in the definitions.

**Comment by tochman** [comment]
Lucia is still working on a task. Not really ready to merge yet

File: app/models/dish.rb [hunk 12]
```diff
@@ -1,11 +1,11 @@
...
-  CATEGORIES = ['Starter', 'Mail Course', 'Dessert']
+  CATEGORIES = ['Starter', 'Main Course', 'Dessert']
   belongs_to :menu
   validates_presence_of :name,
                         :price,
                         :menu,
                         :category
 
-  validates :category, inclusion: ['Starter', 'Mail Course', 'Dessert']
+  validates :category, inclusion: ['Starter', 'Main Course', 'Dessert']
```
Comment (tochman) [inline app/models/dish.rb]:
```
 validates :category, inclusion: CATEGORIES
```

File: features/step_definitions/create_menu_page.rb (hunk 11 above)
Comment (diraulo) [reply features/step_definitions/create_menu_page.rb:19]: Okay then if you have a chore to refactor the step defs.. In general the less code we pull in the first place the less we have to refactor later on. But I will let this one fly for now.

**Comment by diraulo** [comment]
@luciademoja @tochman this one still says WIP?

**Pull request merged**